internal/io/file/file_writer.go

```diff
@@ -71,16 +96,29 @@ func createFileWriter(ctx api.StreamContext, fn string, ft FileType, headers str
 }
 
 func (fw *fileWriter) Close(ctx api.StreamContext) error {
+	var err error
 	if fw.File != nil {
 		ctx.GetLogger().Debugf("File sync before close")
 		_, e := fw.Writer.Write(fw.Hook.Footer())
 		if e != nil {
 			ctx.GetLogger().Errorf("file sink fails to write footer with error %s.", e)
 		}
-		err := fw.Writer.(*bufio.Writer).Flush()
-		if err != nil {
-			ctx.GetLogger().Errorf("file sink fails to flush with error %s.", err)
+		if fw.Compress!="" && fw.Compress!=NONE_COMPRESS {
+			e := fw.Writer.(io.Closer).Close()
+			if e!=nil {
+				ctx.GetLogger().Errorf("file sink fails to close compress writer with error %s.", err)
+			}
+			err = fw.fileBuffer.Flush()
```

Comment:
flush after writer close?

Reply:
I think it's ok here, `fw.Writer` is the compress writer here.